week2/homework/src/index.js

```diff
@@ -1,3 +1,78 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const { readFile, writeFile } = require('fs');
+
+const { promisify } = require('util');
+
+const readFileWithPromise = promisify(readFile);
+const writeFileWithPromise = promisify(writeFile);
+
+const args = process.argv.slice(2);
+
+const cmd = args[0];
+
+let TODO_PATH = 'todo-json';
+
+function readToDos() {
+	return readFileWithPromise(TODO_PATH, 'utf8')
+		.then(JSON.parse)
+		.catch(() => ([]));
+}
+function writeToDos(toDos) {
+	return writeFileWithPromise(TODO_PATH, JSON.stringify(toDos, null, 2));
+}
+
+switch (cmd) {
+	case 'add':
+		const text = args[1];
+		readToDos()
+			.then(toDos => {
+				toDos.push({ toDo: text });
+				return toDos;
+			})
+			.then(writeToDos);
+		break;
+	case 'remove':
+		let index = args[1];
+		readToDos()
+			.then(toDos => {
+				if (index >= 0) {
+					toDos.splice(index, 1);
+				} else {
+					console.info('The index you passed is not valid');
+				}
+				return toDos;
+			}).then(writeToDos);
+		break;
+	case 'list':
+		readToDos()
+			.then(console.log);
+		break;
+	case 'reset':
+		readToDos()
+			.then(toDos => {
+				toDos.splice(0, toDos.length);
+				return toDos;
+			}).then(writeToDos);
+		break;
+	case 'update':
+		const start = args[1];
+		const newText = args[2];
+		readToDos()
+			.then(toDos => {
+				if (start >= 0 && typeof newText === 'string') {
+			toDos.splice(start, 1);
+			toDos[start] = { toDos: newText };
+		}else {
```

Comment:
The indentation here is very confusing. I know you are having problems with your code editor, so maybe this is the reason. We'll have a look at it together tomorrow, but the `else` needs to have the same indentation as the `if`.